app/code/Magento/Sales/Block/Adminhtml/Order/View.php

```diff
@@ -244,11 +245,9 @@ protected function _construct()
             );
         }
 
-        if ($this->_isAllowedAction(
-            'Magento_Sales::reorder'
-        ) && $this->_reorderHelper->isAllowed(
-            $order->getStore()
-        ) && $order->canReorderIgnoreSalable()
+        if ($order->canReorderIgnoreSalable()
```

Comment:
Do we need changing the conditions order? The conditions order matters in that case because first of all, we need to check if the reorder action is allowed. And if it's not, do not perform other checks.

Reply:
Hello @rogyar ,
Thanks for the review, will make suggested changes and update PR.